src/Amadevus.RecordGenerator.Generators/GeneratedCodeAttributeGenerator.cs

```diff
@@ -119,21 +118,9 @@ private AttributeArgumentListSyntax GenerateAttributeArgumentList()
                                 Literal(toolVersion)))}));
         }
 
-        private QualifiedNameSyntax GenerateQualifiedName()
+        private NameSyntax GenerateQualifiedName()
         {
-            var namespaceAndIdentifier = Names.GeneratedCodeAttribute.Split('.')
-                .Select(n => IdentifierName(n)).ToList();
-
-            var attributeName = QualifiedName(
-                namespaceAndIdentifier.First(),
-                namespaceAndIdentifier.Skip(1).First());
-
-            foreach (var identifier in namespaceAndIdentifier.Skip(2))
-            {
-                attributeName = QualifiedName(attributeName, identifier);
-            }
-
-            return attributeName; 
+            return ParseName(Names.GeneratedCodeAttribute);
         }
 
         private string GetToolVersion() => GetType().GetTypeInfo().Assembly.GetName().Version.ToString();
```

Comment:
While refactoring the nearby code, this could be simplified into `ThisAssembly.AssemblyVersion` generated by Nerdbank.GitVersioning, and also then since it's constant, probably inlined completely at usage place.